### ISSUE TITLE
provide way to name fibers

### ISSUE THREAD

**Issue by domdorn** [zio/zio#7658]
I'm having a hard time debugging errors like 

```
"zio.FiberFailure: Interrupted by thread \"zio-fiber-70\"\n\tat com.acme.project.calcu .. 
```

it would be great if we would be able to give fibers a name-prefix to make it easier to find them in the application

**Comment by adamgfraser** [comment]
The name of each fiber includes the location where it is forked from so it is easy to see where that fiber came from.

That being said, I think it could be useful to be able to name fibers at some point just like you can name threads.

One issue is that the current design assumes that the identifier of a fiber is constant throughout its life so I think we would need to provide versions of the various fork operators that allowed specifying a name, which would be unfortunate as there are already probably more variants of fork than we would like.

**Comment by domdorn** [comment]
hmm.. I don't see how `zio-fiber-70` tells me the location where it was forked from?

**Comment by adamgfraser** [comment]
```scala
object Example extends ZIOAppDefault {

  val parent =
    for {
      fiber <- ZIO.never.fork
      exit  <- fiber.interrupt
      _     <- ZIO.debug(exit)
    } yield ()

  val run =
    for {
      fiber <- parent.fork
      _     <- fiber.join
    } yield ()
}
```

```
[info] Failure(Interrupt(Runtime(5,1672864238986,zio.Example.run(Example.scala:14)),Stack trace for thread "zio-fiber-5":
[info]  at zio.Example.parent(Example.scala:8)))
```

**Comment by domdorn** [comment]
doh, I should have copied the whole error above.. 
as far as I remember, the message showed me a generic location, that's why it was not obvious from where it was interrupted and thus I wanted to name the fiber. 

Probably I would need to pass a `implicit t: Trace` or something to the method that created the fiber to get a proper location in the error.
Will cycle back here when I can reproduce the "underlying issue" and give a better example.

**Comment by adamgfraser** [comment]
Okay. The location should always be captured when the fiber is forked.

**Comment by bertlebee** [comment]
> One issue is that the current design assumes that the identifier of a fiber is constant throughout its life so I think we would need to provide versions of the various fork operators that allowed specifying a name, which would be unfortunate as there are already probably more variants of fork than we would like.

Perhaps instead of changing the machinery there could be a name separate to the fiber id, possibly stored as a `FiberRef`. When printing stack traces we could check if that `FiberRef` exists and if so, print it out in stack traces. That also means people don't pay a cost for renaming fibers unless they want to.

**Comment by hearnadam** [comment]
I don't think this will be pursued. FiberId generation should be fast, and it's closely linked with the internals.
